packages/flutter_tools/test/general.shard/flutter_validator_test.dart

```diff
@@ -75,6 +75,49 @@ void main() {
     );
   });
 
+  testWithoutContext('FlutterValidator shows an error message if Rosetta is needed', () async {
+    final FakeFlutterVersion flutterVersion = FakeFlutterVersion(
+      frameworkVersion: '1.0.0',
+      channel: 'beta',
+    );
+    final MemoryFileSystem fileSystem = MemoryFileSystem.test();
+    final Artifacts artifacts = Artifacts.test();
+    final FlutterValidator flutterValidator = FlutterValidator(
+        platform: FakePlatform(
+          operatingSystem: 'macos',
+          localeName: 'en_US.UTF-8',
+          environment: <String, String>{},
+        ),
+        flutterVersion: () => flutterVersion,
+        devToolsVersion: () => '2.8.0',
+        userMessages: UserMessages(),
+        artifacts: artifacts,
+        fileSystem: fileSystem,
+        flutterRoot: () => 'sdk/flutter',
+        operatingSystemUtils: FakeOperatingSystemUtils(name: 'macOS', hostPlatform: HostPlatform.darwin_arm64),
+        processManager: FakeProcessManager.list(<FakeCommand>[
+          const FakeCommand(
+            command: <String>['Artifact.genSnapshot'],
```

Comment:
I'm guessing `Artifacts.test()` always return `this.toString()` when you try to grab the file path?

Reply:
https://github.com/flutter/flutter/blob/8b32ac7a512b35b2b01da949ff41be814d025207/packages/flutter_tools/lib/src/artifacts.dart#L1078-L1096